lib/media/media_source_engine.js

```diff
@@ -586,11 +586,11 @@ shaka.media.MediaSourceEngine = class {
       mimeType = this.transmuxers_[contentType].getOrginalMimeType();
     }
     if (shaka.util.TsParser.probe(uint8ArrayData)) {
-      const metadata = new shaka.util.TsParser().parse(uint8ArrayData)
-          .getMetadata();
+      const tsParser = new shaka.util.TsParser().parse(uint8ArrayData);
+      const timestampOffset =
+          reference.startTime - (tsParser.getStartTime()[contentType] || 0);
```

Comment:
Let me see if I get this.

The SourceBuffer timestampOffset may or may not be set yet, so this is the timestamp offset we would eventually compute for this segment either way.

Is that right?  If so, please add a comment.  The rest LGTM.

Reply:
Yes that's right. I'll add the comment!

Reply:
Done!